configuration.rst

```diff
@@ -914,6 +914,80 @@ parameters at once by type-hinting any of its constructor arguments with the
         }
     }
 
+.. _config-config-builder:
+
+Using PHP ConfigBuilders
+------------------------
+
+Writing PHP config is sometimes difficult because you end up with large nested arrays
+and you have no help from your favorite IDE. A way to address this is to use "ConfigBuilders".
+They are objects that will help you build these arrays.
+
+.. versionadded:: 5.3
+
+    The "ConfigBuilders" was added in Symfony 5.3 as an :doc:`experimental feature </contributing/code/experimental>`.
+
+The ConfigBuilders are automatically generated in your ``kernel.build_dir`` for
+every bundle. By convention they all live in the namespace ``Symfony\Config``.
+
+.. configuration-block::
+
+    .. code-block:: php
+
+        // config/packages/security.php
+        use Symfony\Component\DependencyInjection\Loader\Configurator\ContainerConfigurator;
+
+        return static function (ContainerConfigurator $container) {
+          $array = [
+                'firewalls' => [
+                    'main' => [
+                        'pattern' => '^/*',
+                        'lazy' => true,
+                        'anonymous' => [],
+                    ],
+                ],
+                'access_control' => [
+                    [
+                        'path' => '^/user',
+                        'roles' => 'ROLE_USER',
+                    ],
+                    [
+                        'path' => '^/admin',
+                        'roles' => 'ROLE_ADMIN',
+                    ],
+                ],
+                'role_hierarchy' => [
+                    'ROLE_ADMIN' => ['ROLE_USER'],
+                    'ROLE_SUPER_ADMIN' => ['ROLE_ADMIN', 'ROLE_ALLOWED_TO_SWITCH',
+                    ],
+                ],
+            ];
+
+            $container->extension('security', $array);
+        }
+
+    .. code-block:: ConfigBuilder
```

Comment:
I think it's not supported yet, but it should be "php-config-builder" 

cc @javiereguiluz

Reply:
I was just hoping it would work =)

I make make two code blocks instead. I just wanted to show the differences. 

When Symfony 6 comes around and this (hopefully) is a default feature, we can consider rewriting/updating/adding config blocks.

Reply:
You need to update `_build/config.py` (and Javier needs to do the same on symfony.com).

However: why still document the old array syntax? If this PR is accepted for 5.3, I think we should update all PHP array examples to use this instead of arrays (just like we did with the DI and Routing configurators).